Update Insights Baseline section to include earlier Baseline years

tonypconway opened the pull request:
Hi RUM Archive team!

The data in web-features is now sufficiently complete that you can display features going back to 2016, which is the earliest full year that Edge was available.  This PR tweaks the Baseline section in Insights to do 2 things:

* Show Baseline years back to 2016, not 2017 (might be relevant for folks who are keeping KaiOS 2.5 in mind, which is based on a Firefox/Gecko 48 from 2016)
* Remove the conditional that blocks display of feature groups prior to 2023.

The build process made a lot of small whitespace changes, I'm assuming this isn't a problem.  Let me know if you have any questions :-)

rmarx commented:
Thanks for the PR @tonypconway!

The whitespace stuff does make it supremely annoying to review what has actually changed :) 

At first glance this seems fine, but you missed a few places where I hardcoded to 2017 I think + I first want to make sure the pre-2022 data is actually properly present in the data view I craft in the backend (at https://github.com/rum-archive/rum-insights-data). 

So give me a bit of time :)

tonypconway commented:
Ah good point, missed the _tests.njk file.  I've changed those and run build again.  I can't find any other mentions of 2017 in either repo so I think that's it?  When I previewed it locally, I get great results, i.e. all the features filled in back to 2016:

![Screenshot 2025-04-01 at 11 33 08](https://github.com/user-attachments/assets/98d69fd0-4fbd-4030-8695-d696ea9951fd)

Might it also be worth also updating package.json to a more recent version of web-features?

tonypconway commented:
Hey @rmarx any thoughts on this?  We're currently sourcing things to mention at I/O in May and this might be a good change to feature when we're talking about Baseline tooling.  Let me know if there's anything else I can do.

As an FYI, I've built out [a new function in `baseline-browser-mapping`](https://github.com/web-platform-dx/baseline-browser-mapping#get-data-for-all-browser-versions) that can give you a JS array or CSV of all known browser versions that support some version of Baseline from `@mdn/browser-compat-data` and the modules own downstream version mapping for Yandex, QQ mobile and UC mobile.  Might be a helpful way to bring in more browsers into your stats pipeline.

bluesmoon commented:
@rmarx you can hide whitespace changes in the diff:
![image](https://github.com/user-attachments/assets/f0c09749-5acb-44f4-a672-eaeb157a621b)

bluesmoon reviewed (approved):
Looks good to me.

bluesmoon commented:
@nicjansma I don't think we've ever discussed a contributor license, but I think at the least we'd want to have GPG keys added.  @tonypconway we might need your key ID.

tonypconway commented:
Hey @bluesmoon happy to generate one.  Do you have specific key types/lengths you need?  Haven't created a GPG key before but I can do it now.

tonypconway commented:
I've created a key and added it to my account.  I'll email you all the ID separately.  Do you need me to make another commit once you've added it?

tonypconway commented:
I _think_ that's done, but lmk if you need something else.

bluesmoon commented:
Looks like it worked... the last 2 commits are marked as verified, but the first 4 are not.  You'll need to squash (git rebase -i) to merge those commits into one and make a new single verified commit (or any other way you want to redo those commits).

bluesmoon commented:
![image](https://github.com/user-attachments/assets/cf2262b1-8364-4505-8485-671c6a0183b5)

tonypconway commented:
OK, I think that does it.  Single commit, verified, `feed.xml` updated to today.  All good?  Sorry, relatively new to rebasing, but I think I've got my head around it now.

tonypconway commented:
Should be ok now:

<img width="1266" alt="image" src="https://github.com/user-attachments/assets/f3315519-81f9-4d34-85ac-9632b4da8cae" />

Changed laptops as I'm at home now and didn't have GPG verification fully set up on this one.  Rebased again.
